src/endpoints/v1.es6.js

```diff
@@ -461,6 +461,54 @@ class APIv1Endpoint {
             return [];
           }
         });
+      },
+
+      post: function(options = {}) {
+        var uri = options.origin + '/api/submit';
+
+        if (!options.model) {
+          throw new NoModelError('/api/submit');
+        }
+
+        var valid = options.model.validate();
+
+        if (valid === true) {
+          var json = options.model.toJSON();
+
+          options.form = {
+            api_type: 'json',
+            thing_id: json.thingId,
+            title: json.title,
+            kind: json.kind,
+            sendreplies: json.sendreplies,
+            sr: json.sr,
+            iden: json.iden,
+            captcha: json.captcha,
+            resubmit: json.resubmit,
+          };
+
+          if (json.text) {
+            options.form.text = json.text;
+          } else if (json.url) {
+            options.form.url = json.url;
+          }
+
+          return basePost(uri, options, this.request, (body) => {
+            if (body.json && body.json.errors.length < 1) {
+              // TODO handle errors
+              if (!body.json.errors.length) {
+                return body.json.data
+              }
+              // var link = body.json.data.things[0].data;
+            } else {
+              return body.json.errors;
```

Comment:
When you have a moment could you show me how to do this? It's not calling the reject function.